docs/contributing-to-airbyte/building-new-connector/testing-connectors.md

```diff
@@ -1,16 +1,80 @@
+_Sequence diagram of a Standard Test Suite run_. 
+
+The Standard Test Suite runs its tests against the connector's Docker image. It takes as input the following:
+* **The Connector's Docker image name**, so it can run tests against that image
+* **A config file** containing necessary credentials/information to connect to the underlying data source/destination
+* **A configured catalog** that will be used when running read/write operations against the connector
+* **(Optional) A state file** for use in incremental sync test scenarios
+
+The test suite then runs its test cases, which include: 
+* Using the input config file, running the `check` operation should succeed. 
+* Using a made up/fake config file, running the `check` operation should fail. 
+* Running a `read` operation should produce at least one record. 
+* Running two consecutive full refresh reads should produce identical records. 
+etc...
+
+See all the test cases and their description in [Standard Source Tests](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/StandardSourceTest.java). 
+All test cases are annotated with the `@Test` annotation. 
+
+### Setting up standard tests for your connector
+Standard tests are typically run from a docker container. The default standard test runner is the File-based Standard Test suite, which
+gets its name because its inputs are passed as files via Docker volume mounts. This is the simplest way to run the standard test
+suite: the only requirements are that you place its input files inside your connector's directory, and to pass the paths to those input
+files as arguments to the Gradle plugin required to invoke it. This is setup by default inside the `build.gradle` file on all
+connectors generated from templates. 
+  
+For reference,to configure the file-based standard test suite the only requirement is to add the following block in your connectors 
+`build.gradle` file:
+```
+apply plugin: 'airbyte-standard-source-test-file'
+airbyteStandardSourceTestFile { 
+  // all these paths must be inside your connector's directory
+  configPath = "/path/to/config"
+  specPath = "/path/to/spec"
+  configuredCatalogPath = "/path/to/catalog"
+}
+```
+
+These inputs are all described in the [Airbyte Specification](../../architecture/airbyte-specification.md) and will be used as follows: 
+* **Spec file** will be compared to the spec file output by the connector when the `spec` command is called. 
+* **Config file** is expected to be a valid config file. It's expected that calling `check` with this config will succeed. 
+* **Configured Catalog** read operations will be performed on all the streams found in this catalog. All sync modes supported for each stream will be tested. If any stream requires a user-defined cursor, this should be defined in this catalog file. (If this sounds like gibberish, make sure to read about incremental sync). 
```

Comment:
TODO once I add docs for incremental sync i'll link to them in the part that says "make sure to read about incremental sync"